target/linux/sifiveu/image/Makefile

```diff
@@ -0,0 +1,63 @@
+# SPDX-License-Identifier: GPL-2.0-only
+#
+# Copyright (C) 2022 Toco Technologies <info@toco.ae>
+#
+include $(TOPDIR)/rules.mk
+include $(INCLUDE_DIR)/image.mk
+
+FAT32_BLOCK_SIZE=1024
+FAT32_BLOCKS=$(shell echo $$(($(CONFIG_SIFIVEU_SD_BOOT_PARTSIZE)*1024*1024/$(FAT32_BLOCK_SIZE))))
+
+KERNEL_LOADADDR:=0x80200000
+
+define Build/riscv-sdcard
+	rm -f $@.boot #$(KDIR_TMP)/$(IMG_PREFIX)-$(PROFILE)-boot.img
+	mkfs.fat $@.boot  -C $(FAT32_BLOCKS)
```

Comment:
As we are building U-Boot from source, why use a filesystem from the 1980's to store the kernel?
The boot script be embedded in the default environment of the U-Boot build and uImage.FIT can be stored directly in a partition. In this way the bootloader can validate everything needed for a successful boot before launching the kernel (ie. kernel, dtb **and rootfs**) and reliably dual-boot or launch recovery mechanics in case of integrity errors while avoiding the single point of failure of the FAT filesystem. It also allows to use a single type of image (uImage.FIT) equally on NOR/raw MTD, NAND/UBI and eMMC/block devices.
Examples:
https://github.com/openwrt/openwrt/blob/master/target/linux/mediatek/image/mt7622.mk#L98
https://github.com/openwrt/openwrt/blob/master/package/boot/uboot-mediatek/patches/404-add-bananapi_bpi-r64_defconfigs.patch#L164

I understand that in SBC distributions this aspect is usually neglected as the assumption is that you can always just remove the SD card and stick it into your (Windows) computer. While this is nice to keep the entrance barrier low for tinkerers, the requirements for OpenWrt often look a bit different and involve devices out of the physical reach of their operators...